### PR TITLE
Rename ServoServer to ServoNode

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

Until we figure out how we want to change the API to prevent this sort of bug this makes servo error much nicer in the constructor of ServoCalcs.

This also includes a fix for a clang-tidy error I hadn't seen before.  I wonder if our clang-tidy changed since the last time someone made a code change to servo.

Updated tutorials: https://github.com/ros-planning/moveit2_tutorials/pull/109

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [x] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers
- [x] Make PR to update tutorials.
- [x] Migration note

### CODE REVIEW

File: moveit_ros/moveit_servo/src/teleop_demo/joystick_servo_example.cpp
Comment (henningkayser) [inline moveit_ros/moveit_servo/src/teleop_demo/joystick_servo_example.cpp:52]: Should we just use the `/servo/*` namespace?

File: moveit_ros/moveit_servo/include/moveit_servo/servo_component.h
Comment (henningkayser) [inline moveit_ros/moveit_servo/include/moveit_servo/servo_component.h:48]: What are the implications of calling this class Component? Isn't it a node? So `ServoNode` makes most sense to me, which in turn can be a component node (exported, no class/file needed) or a node executable `ServoNodeExecutable`.

File: moveit_ros/moveit_servo/CMakeLists.txt
Comment (AndyZe) [inline moveit_ros/moveit_servo/CMakeLists.txt:19]: I like this name a lot more

File: moveit_ros/moveit_servo/include/moveit_servo/servo_component.h
Comment (AndyZe) [reply moveit_ros/moveit_servo/include/moveit_servo/servo_component.h:48]: Changes here should be reviewed along with #621 (Add standalone executable for Servo server node). In fact, maybe this PR should be closed and we just make all of the naming changes as a separate commit in that PR.

I think `ServoNodeComponent` (class name) and `servo_component_node.cpp` make good sense for this class / file.

The executable that Joe added in #621 could be called `ServoMain` or `ServoStandaloneNode`. It just instantiates this class.

File: moveit_ros/moveit_servo/src/teleop_demo/joystick_servo_example.cpp
Comment (tylerjw) [reply moveit_ros/moveit_servo/src/teleop_demo/joystick_servo_example.cpp:52]: agree that probably makes sense or /moveit_servo ?

Comment (AndyZe) [reply moveit_ros/moveit_servo/src/teleop_demo/joystick_servo_example.cpp:52]: :+1: to /servo/*

File: moveit_ros/moveit_servo/include/moveit_servo/servo_component.h
Comment (schornakj) [reply moveit_ros/moveit_servo/include/moveit_servo/servo_component.h:48]: +1 to `ServoNode`. In my mind we create a node and then allow launching it in a component container.

**Comment by tylerjw** [comment]
I renamed servo_component to servo_node and servo_server_node to servo_node_main.

**Review by henningkayser — APPROVED** [review]

**Comment by henningkayser** [comment]
@tylerjw one thing before merging. Could you provide deprecated headers and typedefs for the old API?

**Comment by tylerjw** [comment]
@AndyZe I believe this is ready for review.  I included a deprecated header so existing code should continue to build and produce a warning.  There is no way for me to deprecate the change in the name for the binaries (the component node plugin) so launch files that depend on the old ServoServer node name will stop working after this.

File: moveit_ros/moveit_servo/include/moveit_servo/servo_server.h
Comment (JafarAbdi) [inline moveit_ros/moveit_servo/include/moveit_servo/servo_server.h:43]: Should we add a date for when this header and the alias will be removed, I like how drake does it https://github.com/RobotLocomotion/drake/blob/master/common/drake_deprecated.h#L59

**Review by JafarAbdi — APPROVED** [review]

File: moveit_ros/moveit_servo/include/moveit_servo/servo_server.h
Comment (henningkayser) [reply moveit_ros/moveit_servo/include/moveit_servo/servo_server.h:43]: I don't this is necessary. We'll remove it with the next major (=API breaing) release. No need to set a date for it

**Review by AndyZe — APPROVED** [review]
A bit of text still needs to be replaced. Approved assuming that happens.

```
moveit_ros/moveit_servo/launch/servo_example.launch.py:            #     name="servo_server",
moveit_ros/moveit_servo/launch/servo_example.launch.py:        executable="servo_server_node",
```

File: moveit_ros/moveit_servo/include/moveit_servo/servo_server.h
Comment (tylerjw) [reply moveit_ros/moveit_servo/include/moveit_servo/servo_server.h:43]: I think we should just plan on deleting all the deprecated APIs right before the next major release, Humble.  We can of course do it sooner if we think that is necessary for some other reason.  main/Rolling are not guaranteed to be API stable but we should attempt to not break it often if we can.

**Comment by tylerjw** [comment]
> It looks like this PR along with some of the other recent changes (e.g. launch file shuffling) have broken the Servo tutorial. I'd like to put the brakes on more large Servo changes until the tutorial gets fixed.

The tutorial on the website or the tutorial in the main branch of the tutorial repo?  The website has changed to track a new foxy branch.  That website is based on a foxy install of moveit.

**Comment by AndyZe** [comment]
main branch, built from source

**Comment by AndyZe** [comment]
Yep, the plan for the tutorials makes sense. We should try to keep the `main` tutorials branch up to date with Rolling, though. Partly my fault for approving this PR without testing it.

**Pull request merged**
